server/middlewares/UserValidation.js

```diff
@@ -16,20 +17,65 @@ class UserValidation {
     * @static
     */
   static validateUserSignUp(req, res, next) {
-    let { fullName } = req.body;
-    const { avatarUrl, bio } = req.body;
-    req.checkBody('fullName', 'full name is too long').isLength({ max: 40 });
+    UserValidation.validateFullName(req);
+    UserValidation.validateEmail(req);
+    UserValidation.validatePassword(req);
+    UserValidation.validateAvatarUrl(req);
+    UserValidation.validateBio(req);
+    UserValidation.sendFormattedError(req, res, next);
+  }
+
+  /**
+    * @description - This method validates the email
+    * @param {object} req - The request object
+    * @returns {null} - returns nothing
+    * @memberOf UserValidation
+    * @static
+    */
+  static validateEmail(req) {
     req.checkBody('email', 'please enter email').exists();
     req.checkBody('email', 'please enter a valid email').isEmail();
+  }
+
+  /**
+    * @description - This method validates the password
+    * @param {object} req - The request object
+    * @param {object} password - The request object
```

Comment:
this `password` argument doesn't exist on this function, I don't think you meant to put it here.

Reply:
Thanks for the feedback. I have taken note.